src/drivers/fluid_oboe.cpp

```diff
@@ -0,0 +1,218 @@
+  int32_t cont;
+  OboeAudioStreamCallback *oboe_callback;
+  AudioStream *stream;
+} fluid_oboe_audio_driver_t;
+
+
+class OboeAudioStreamCallback : public AudioStreamCallback
+{
+public:
+
+  OboeAudioStreamCallback (void *userData)
+    : user_data (userData)
+  {
+  }
+
+  void *user_data;
+
+  DataCallbackResult onAudioReady (AudioStream *stream, void *audioData, int32_t numFrames)
+  {
+    float *callback_buffers[2];
+    fluid_oboe_audio_driver_t *dev;
+    OboeAudioStreamCallback *oboe_callback;
+  
+    dev = (fluid_oboe_audio_driver_t*) user_data;
+  
+    if (!dev->cont)
+      return DataCallbackResult::Stop;
+  
+    if (stream->getFormat () == AudioFormat::Float)
+    {
+      fluid_synth_write_float(dev->synth, numFrames, (float*) audioData, 0, 2, (float*) audioData, 1, 2);
+    }
+    else
+    {
+      fluid_synth_write_s16(dev->synth, numFrames, (short*) audioData, 0, 2, (short*) audioData, 1, 2);
+    }
+    return DataCallbackResult::Continue;
+  }
+};
+
+void fluid_oboe_audio_driver_settings(fluid_settings_t* settings)
+{
+  fluid_settings_register_int(settings, "audio.oboe.device-id", 0, 0, 0x7FFFFFFF, 0);
+  
+  fluid_settings_register_str(settings, "audio.oboe.sharing-mode", "Shared", 0);
+  fluid_settings_add_option(settings,   "audio.oboe.sharing-mode", "Shared");
+  fluid_settings_add_option(settings,   "audio.oboe.sharing-mode", "Exclusive");
+  
+  fluid_settings_register_str(settings, "audio.oboe.performance-mode", "None", 0);
+  fluid_settings_add_option(settings,   "audio.oboe.performance-mode", "None");
+  fluid_settings_add_option(settings,   "audio.oboe.performance-mode", "PowerSaving");
+  fluid_settings_add_option(settings,   "audio.oboe.performance-mode", "LowLatency");
+}
+
+
+/*
+ * new_fluid_oboe_audio_driver
+ */
+fluid_audio_driver_t*
+new_fluid_oboe_audio_driver(fluid_settings_t* settings, fluid_synth_t* synth)
+{
+  Result result;
+  fluid_oboe_audio_driver_t* dev;
+  AudioStreamBuilder builder_obj;
+  AudioStreamBuilder *builder = &builder_obj;
+  AudioStream *stream;
+  
+  int period_frames;
+  double sample_rate;
+  int is_sample_format_float;
+  int device_id;
+  int sharing_mode; // 0: Shared, 1: Exclusive
+  int performance_mode; // 0: None, 1: PowerSaving, 2: LowLatency
+
+  try {
+
+  dev = FLUID_NEW(fluid_oboe_audio_driver_t);
+  if (dev == NULL) {
+    FLUID_LOG(FLUID_ERR, "Out of memory");
+    return NULL;
+  }
+
+  FLUID_MEMSET(dev, 0, sizeof(fluid_oboe_audio_driver_t));
+  
+  dev->synth = synth;
+  dev->oboe_callback = new (std::nothrow) OboeAudioStreamCallback(dev);
+  if (!dev->oboe_callback) {
+    FLUID_LOG(FLUID_ERR, "Out of memory");
+    goto error_recovery;
+  }
+
+  fluid_settings_getint(settings, "audio.period-size", &period_frames);
+  fluid_settings_getnum(settings, "synth.sample-rate", &sample_rate);
+  is_sample_format_float = fluid_settings_str_equal (settings, "audio.sample-format", "float");
+  fluid_settings_getint(settings, "audio.oboe.device-id", &device_id);
```

Comment:
When going through the other settings I realized that `audio.oboe.device-id` might be a little inconsistent. Usually we have settings like `audio.NAME.device` as string. Since oboe receives an integer I'd suggest to rename it to `audio.oboe.id`. I know we have `audio.jack.id` being already inconsistent as it receives a string, but `audio.oboe.id` seems to be the better choice in my opinion.

Reply:
Alright, if you say so :-)